staging/src/k8s.io/client-go/applyconfigurations/admissionregistration/v1/mutatingwebhook.go

```diff
@@ -0,0 +1,256 @@
+)
+
+// MutatingWebhookApplyConfiguration represents an declarative configuration of the MutatingWebhook type for use
+// with apply.
+type MutatingWebhookApplyConfiguration struct {
+	fields mutatingWebhookFields
+}
+
+// MutatingWebhookApplyConfiguration constructs an declarative configuration of the MutatingWebhook type for use with
+// apply.
+func MutatingWebhook() *MutatingWebhookApplyConfiguration {
+	return &MutatingWebhookApplyConfiguration{}
+}
+
+// mutatingWebhookFields owns all fields except inlined fields.
+// Inline fields are owned by their respective inline type in MutatingWebhookApplyConfiguration.
+// They are copied to this type before marshalling, and are copied out
+// after unmarshalling. The inlined types cannot be embedded because they do
+// not expose their fields directly.
+type mutatingWebhookFields struct {
+	Name                    *string                                         `json:"name,omitempty"`
+	ClientConfig            *WebhookClientConfigApplyConfiguration          `json:"clientConfig,omitempty"`
+	Rules                   *RuleWithOperationsList                         `json:"rules,omitempty"`
+	FailurePolicy           *admissionregistrationv1.FailurePolicyType      `json:"failurePolicy,omitempty"`
+	MatchPolicy             *admissionregistrationv1.MatchPolicyType        `json:"matchPolicy,omitempty"`
+	NamespaceSelector       *metav1.LabelSelectorApplyConfiguration         `json:"namespaceSelector,omitempty"`
+	ObjectSelector          *metav1.LabelSelectorApplyConfiguration         `json:"objectSelector,omitempty"`
+	SideEffects             *admissionregistrationv1.SideEffectClass        `json:"sideEffects,omitempty"`
+	TimeoutSeconds          *int32                                          `json:"timeoutSeconds,omitempty"`
+	AdmissionReviewVersions *[]string                                       `json:"admissionReviewVersions,omitempty"`
+	ReinvocationPolicy      *admissionregistrationv1.ReinvocationPolicyType `json:"reinvocationPolicy,omitempty"`
+}
+
+// SetName sets the Name field in the declarative configuration to the given value.
+func (b *MutatingWebhookApplyConfiguration) SetName(value string) *MutatingWebhookApplyConfiguration {
+	b.fields.Name = &value
+	return b
+}
+
+// Name gets the Name field from the declarative configuration.
+func (b *MutatingWebhookApplyConfiguration) Name() (value string, ok bool) {
+	if v := b.fields.Name; v != nil {
+		return *v, true
+	}
+	return value, false
+}
+
+// SetClientConfig sets the ClientConfig field in the declarative configuration to the given value.
+func (b *MutatingWebhookApplyConfiguration) SetClientConfig(value *WebhookClientConfigApplyConfiguration) *MutatingWebhookApplyConfiguration {
+	b.fields.ClientConfig = value
+	return b
+}
+
+// ClientConfig gets the ClientConfig field from the declarative configuration.
+func (b *MutatingWebhookApplyConfiguration) ClientConfig() (value *WebhookClientConfigApplyConfiguration, ok bool) {
```

Comment:
is it still possible to chain getters because of the multi-return value?

Reply:
I'm curious if that experience matches what was described in the KEP? Also, I understand that you always need to do nil pointer checks, so there's no way around that.

Reply:
That's a good point. I remember we had an integration test in the go structs branch to exercise these conditions. I couldn't seem to find it in this branch, but it would probably be good to define the desired behavior and show an example. This does seem like a somewhat difficult thing to address though.

Reply:
We (@apelisse, @Jefftree, @kevindelgado) had an offline discussion about this. Consensus was: We will expose the underlying structs and only generate setter functions. This sidesteps the problems that come up with the getters like what they should return since there were downsides to each alternative we discussed of what we have the getters return.